src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -38,6 +45,78 @@ public static void ProcessLogFile(string filePath, Solution solution, string too
             if (toolFormat.MatchesToolFormat(ToolFormat.None))
             {
                 logText = File.ReadAllText(filePath);
+
+                if (logText.Contains(@"""version"": ""1.0.0"""))
+                {
+                    // They're opening a v1 log, so we need to transform it
+                    // Ask if they'd like to save the v2 log
+                    int response = VsShellUtilities.ShowMessageBox(SarifViewerPackage.ServiceProvider,
+                                                                   Resources.TransformV1_DialogMessage,
+                                                                   null, // title
+                                                                   OLEMSGICON.OLEMSGICON_QUERY,
+                                                                   OLEMSGBUTTON.OLEMSGBUTTON_YESNOCANCEL,
+                                                                   OLEMSGDEFBUTTON.OLEMSGDEFBUTTON_FIRST);
+
+                    if (response == IDCANCEL)
```

Comment:
1. Why "yes/no/cancel" as opposed to just "yes/no"?
2. If you do use "yes/no/cancel" then you need to return on `IDNO`, too.